apps/balancer-tools/src/app/apr/round/(components)/TopPoolsChart.tsx

```diff
@@ -10,54 +8,60 @@ import {
 import { PlotType } from "plotly.js";
 
 import Plot from "#/components/Plot";
+import votingGauges from "#/data/voting-gauges.json";
+
+import { calculatePoolStats } from "../../(utils)/calculateRoundAPR";
+import sortDataByX from "../../(utils)/sortChartData";
 
-export default function TopPoolsChart({ roundId }: { roundId: string }) {
+export default async function TopPoolsChart({ roundId }: { roundId: string }) {
+  const POOL_QUANTITY_LIMIT = 16;
   const colors = [
     greenDark.green9,
     violetDark.violet9,
     crimsonDark.crimson8,
     blueDark.blue9,
     amberDark.amber9,
   ];
-  const bars = [
-    { x: "0.00", y: "sNOTE-BPT" },
-    { x: "0.01", y: "B-50USDC-50WETH" },
-    { x: "0.35", y: "B-stETH-STABLE" },
-    { x: "0.50", y: "B-50VITA-50WETH" },
-    { x: "0.99", y: "80D2D-20USDC" },
-    { x: "1.10", y: "B-80GNO-20WETH" },
-    { x: "1.92", y: "50WETH-50AURA" },
-    { x: "3.48", y: "20WBTC-80BADGER" },
-    { x: "3.53", y: "50COW-50GNO" },
-    { x: "5.60", y: "50COW-50WETH" },
-    { x: "9.60", y: "VBPT" },
-    { x: "18.19", y: "20WETH-80WNCG" },
-    { x: "18.62", y: "B-sdBAL-STABLE" },
-    { x: "26.18", y: "50DFX-50WETH" },
-    { x: "39.03", y: "USDC-PAL" },
-  ];
-  const colorArray = Array.from(
-    { length: bars.length },
-    (_, index) => colors[index % colors.length],
-  );
 
-  const data = {
+  const barCords: {
+    x: string[];
+    y: string[];
+    colors: string[];
+  } = { x: [], y: [], colors: [] };
+  for (let index = 0; index < votingGauges.length; index++) {
+    if (barCords.x.length >= POOL_QUANTITY_LIMIT) {
+      break;
+    }
+
+    const gauge = votingGauges[index];
+    const { apr } = await calculatePoolStats({
+      roundId,
+      poolId: gauge.pool.id,
+    });
+
+    if (apr && apr > 0.01) {
+      barCords.x.push(apr.toFixed(2));
+      barCords.y.push(gauge.pool.symbol);
+      barCords.colors.push(colors[index % colors.length]);
+    }
```

Comment:
Some strange behavior is happening here, you are a limit of 16 pools but only 13 are being rendered. Probably because not every pool from that 16 has APR. Might be something that #226  will help

Reply:
Probably due to the `apr > 0.01` logic, it doesn't necessarily mean that it will display 16; it only displays a maximum of 16. However, I do think it's a valid point that displaying a fixed number is more user-friendly.